pydoctor/model.py

```diff
@@ -78,6 +78,34 @@ class PrivacyClass(Enum):
     PRIVATE = 1
     VISIBLE = 2
 
+class _KindEnum(Enum):
+    """
+    Abstract container for L{KindClass} values. 
+    """
+    @property
+    def name(self) -> str: # type: ignore[override]
+        return super().name.replace("_", " ")
+
+class KindClass(_KindEnum):
+    """
+    Values indicating the possible object types.
+
+    @note: Some new kinds can be dynamically created by using `zope.schema`.
```

Comment:
I don’t see other way to deal with zope schema thing. We need to have dynamic kinds names. I don’t understand what you mean by being just a shortcut. In any case, this pattern is sketchy... and I don’t see how we could meet the requirements with static enum values.

Reply:
I have never used zope schemas, only interfaces, so I'll have to dive into the zope docs to figure out what this is even for.

Reply:
With "shortcut" I meant that when the original author decided to create new `kind` values dynamically, I wonder if that was a deliberate design decision or just the easiest way to implement it at the time and that initial implementation stuck around.

Reply:
Just look at our tests regarding zope.schema. No idea how we could reproduce this behaviour with a fix amount of enum values. 

https://github.com/twisted/pydoctor/blob/e32d0bac24d4c901454ec076becb031fa834bd15/pydoctor/test/test_zopeinterface.py#L152

Reply:
I guess the right way to do it would be to list all possible schemas here: https://zopeschema.readthedocs.io/en/latest/api.html#schema-apis

But that's definitely overkill in my opinion, I mean who is using this anyway ? 

I think we can go with a new kind value like `ZOPE.SCHEMA_FIELD`. Then we could rather store the type information inside the `parsed_type` attribute ? (It's currently `None`) It makes more sens to me anyway. 

What think ye ?

Reply:
`zope.schema` looks similar to `attr.ib` from `attrs`. The schema class is more or less the type of the attribute, so I think `kind` wasn't really the best place to store the schema class to begin with.

Having a dedicated `SCHEMA_FIELD` kind would be an option indeed. Or maybe we could use the existing `ATTRIBUTE` kind.

I'm not sure what the criteria are for defining "kinds". An attribute is an instance variable, but we have it as a separate kind and I'm not sure why. Similarly, there is #364 which asks for a separate kind for exceptions.

There is some connection between the `Documentable` subclass and the `kind`, in that for example a documentable of kind `METHOD` is always an instance of `Function`. But there can be more than one kind of each subclass: for `Function` there is also `CLASS_METHOD`, `STATIC_METHOD` and plain `FUNCTION`. And in some cases it's not even a 1:1 mapping, for example a property is implemented using 1 to 3 methods (get/set/del), which isn't handled all that well at the moment.

I don't think the `kind` is used much for analysis, it seems to be mainly a presentation thing. So perhaps it doesn't even belong in the model. So we might want to overhaul the design in a more thorough way, but I think that it's best to first finish the conversion from string to enum and merge this PR before we start making bigger changes.

To come back to the current issue: I don't think we format schema fields in any particular way, except that we display the `kind` field itself. I don't know of any project that uses `zope.schema` (Twisted doesn't use it), so I wouldn't know what works better in practice. So I'll leave it up to you to pick between `SCHEMA_FIELD` and `ATTRIBUTE`, whichever is easier. If a user disagrees they can file a ticket and we make it nicer then; we'll then have someone that can show us real use cases.

Reply:
Ok I've pushed what I think is the right thing to do: Using AnnotationDocstring to link to zope.schema type in the apidocs.